tempesta_fw/http_parser.c

```diff
@@ -866,6 +850,10 @@ __FSM_STATE(RGen_BodyChunkExt) {					\
 	return TFW_BLOCK;						\
 }
 
+#define TFW_HTTP_UPDATE_BODY_IF_OPENED()				\
+	if (msg->body.ptr)						\
+		tfw_http_msg_add_data_ptr(msg, &msg->body, data, len);
```

Comment:
I think the macro is redundant. At any FSM state of HTTP message chunk processing we have `data` pointing to begin of the chunk. So we can set up the body chunk at any time. The macro is especially bad for HTTP requests which usually don't have body at all, but we still do the branching (BTW it would be a good place to use `unlikely`)